### PR TITLE
CI script to check template fields

### PR DESCRIPTION
**Pull request by gopidesupavan**
Adding CI script to check template fields in provider operators and sensors

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by gopidesupavan** [comment]
cc: @ferruzzi 

A recent update to the #41848 template field caused a system test failure. To prevent similar issues, I started adding tests to verify all template fields. As suggested by ferruzzi in the discussion https://github.com/apache/airflow/pull/42183#issuecomment-2347025162, implementing a CI check would be a valuable addition to this process.

**Comment by gopidesupavan** [comment]
@potiuk Could you pleas help where to configure this script ?, Not sure how to configure this to CI level :) , I tried using a pre-commit hook, but it's encountering dependency issues. I've moved the script to a in_container to resolve these dependencies. or could you please direct me where to place this?

**Comment by gopidesupavan** [comment]
cc: @o-nikolas

**Comment by gopidesupavan** [comment]
Have another question at present it reads the providers yaml files and validates all the template fields from operators/sensors, is it good idea to check all the providers everytime part of pre-commit ? or only providers changed part of that commit? Please suggest, what would be the best approach?

**Comment by gopidesupavan** [comment]
Am able to figure-out how to configure in pre-commit script :).

**Review by vincbeck — APPROVED** [review]

**Review by o-nikolas — COMMENTED** [review]
I assume you manually tested this an it's all working as expected?

File: scripts/ci/pre_commit/check_template_fields.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    backend="sqlite",
+    warn_image_upgrade_needed=True,
+    extra_env={"PYTHONWARNINGS": "default"},
+)
+if cmd_result.returncode != 0 and os.environ.get("CI") != "true":
+    console.print(
+        "\n[yellow]If you see strange stacktraces above, especially about missing imports "
+        "run this command:[/]\n"
+    )
+    console.print("[magenta]breeze ci-image build --python 3.8 --upgrade-to-newer-dependencies[/]\n")
```
Comment (o-nikolas) [inline scripts/ci/pre_commit/check_template_fields.py]: We have this in a few places. Can we not just pop this into `run_command_via_breeze_shell` itself?

**Comment by gopidesupavan** [comment]
> I assume you manually tested this an it's all working as expected?

Yes , sorry i would have posted some screenshots :) 

image

I hope this is fine?

File: scripts/ci/pre_commit/check_template_fields.py (hunk 1 above)
Comment (gopidesupavan) [reply scripts/ci/pre_commit/check_template_fields.py]: correct, let me check if am able to move some of these kind blocks.

**Review by ferruzzi — APPROVED** [review]
:heart_eyes: 

Other than Niko's suggestion for moving some duplicate code, and the need to rebase, I love it.   Thanks for taking care of this!


I saw you tagged Jarek with a question; he's going to be away for a couple weeks.

File: scripts/ci/pre_commit/check_template_fields.py (hunk 1 above)
Comment (gopidesupavan) [reply scripts/ci/pre_commit/check_template_fields.py]: Updated, I hope this is fine? moved these checks inside to common pre-commit utils.

**Comment by gopidesupavan** [comment]
> 😍
> 
> Other than Niko's suggestion for moving some duplicate code, and the need to rebase, I love it. Thanks for taking care of this!
> 
> I saw you tagged Jarek with a question; he's going to be away for a couple weeks.

Thank you 😄

**Comment by gopidesupavan** [comment]
Can i get some help please? tests are failing , not sure exactly is it due to any intermittent issues in pipeline.

**Comment by gopidesupavan** [comment]
Sorry i messed up something while rebasing

**Comment by gopidesupavan** [comment]
Is it okay if i create new clean PR with referencing this one, due to some issues while doing rebase all the previous merged commits again showing in changes 😞 .

**Comment by dstandish** [comment]
> Is it okay if i create new clean PR with referencing this one, due to some issues while doing rebase all the previous merged commits again showing in changes 😞 .

go for it

**Comment by gopidesupavan** [comment]
> > Is it okay if i create new clean PR with referencing this one, due to some issues while doing rebase all the previous merged commits again showing in changes 😞 .
> 
> go for it

Thank you. created new pr.

**Pull request closed without merging**